src/client_side_request.cc

```diff
@@ -1095,7 +1098,10 @@ clientInterpretRequestHeaders(ClientHttpRequest * http)
 
 #endif
 
-    request->flags.cachable = http->request->maybeCacheable();
+    if (http->request->maybeCacheable())
+        request->flags.cachable.support();
+    else
+        request->flags.cachable.veto();
```

Comment:
We can remove/simplify the above `if` statement as illustrated below, but I am worried that folks will start adding `update(false)` and `update(true)` calls elsewhere if we add such an update() method. Compared to explicitly calling support() and veto(), calling update() decreases code readability, especially since the difference between true and false goes beyond pure boolean algebra here (due to veto powers of negative votes). Thus, I do not think we should add update() (or a similar method).

```C++
         request->flags.cachable.update(http->request->maybeCacheable());
```